libminifi/include/properties/Configure.h

```diff
@@ -42,12 +44,23 @@ class Configure : public Configuration, public core::AgentIdentificationProvider
   std::string getAgentIdentifier() const override;
   void setFallbackAgentIdentifier(const std::string& id);
 
+  using Configuration::set;
+  void set(const std::string& key, const std::string& value, PropertyChangeLifetime lifetime) override;
+  bool commitChanges() override;
+
+  // WARNING! a test utility
+  void setLoggerProperties(std::shared_ptr<core::logging::LoggerProperties> new_properties) {
+    logger_properties_ = new_properties;
+  }
```

Comment:
If this is only for testing could this be made private and declare the test class be a friend class?

Reply:
done